src/features/wallet/helpers/checkAvailableRewards.ts

```diff
@@ -0,0 +1,27 @@
+import { ethers } from "ethers";
+import { VoteRevealed } from "web3/get/queryVotesRevealedEvents";
+import { queryRetrieveRewards } from "web3/get/queryRetrieveRewards";
+
+export default async function checkAvailableRewards(
+  data: VoteRevealed[],
+  address: string,
+  contract: ethers.Contract
+) {
+  const promises = data.map(async (vote) => {
+    const rewardAvailable = await queryRetrieveRewards(
```

Comment:
Minor note (not sure if there's anything to change): it looks like we're casting this to a float then using it to determine _if_ there are any rewards to claim. If we have rewards that are so small that they are below the precision limit of JS floats, we may accidentally truncate nonzero rewards to 0 and think that the user has _no_ rewards to claim when they really have just a very small number. I'm not sure how small their initial balance would need to be for this to happen, but just wanted to mention it since this float seems to be used for more than just display.

Reply:
i noticed this as well, and you bring up a really good edge case. I think in general we should default to handling these numbers as wei strings/bn's until its time to display / math them. and maybe define some guidelines around how to deal with BNs across all our apps since its tricky.